bayesian_changepoint_detection_experiment/01_change_point_detection.py

```diff
@@ -48,52 +50,88 @@
         word_freq_df = pd.DataFrame.from_records(
             [
                 {
-                    "tok": tok,
+                    "tok": re.escape(tok),
                     "word_count": wv_model.wv.get_vecattr(tok, "count"),
                     "year": re.search(r"(\d+)_0", str(model)).group(1),
                 }
                 for tok in wv_model.wv.index_to_key
             ]
         )
-        total_count = word_freq_df.word_count.sum()
+
+        total_count = (word_freq_df >> ply.pull("word_count")).sum()
+
         word_freq_df = word_freq_df >> ply.define(frequency=f"word_count/{total_count}")
+
         data.append(word_freq_df)
 
+# +
 if not Path("output/all_tok_frequencies.tsv.xz").exists():
-    all_word_freq_df = pd.concat(data)
-    all_word_freq_df.dropna().to_csv(
-        "output/all_tok_frequencies.tsv.xz", sep="\t", index=False, compression="xz"
+    all_word_freq_df = pd.concat(data) >> ply.call(".dropna")
```

Comment:
>cool new package! So just for my own understanding is the this pipe-ing equivalent to pd.concat(data).dropna().to_csv()? Is there a reason to use one over the other?

Yup! Its called plydata. Mimics R's dplyr package. Main reason I switched is because stylistically my code became easier to read and comprehend. The periods are good for pandas but I like the pipe-ing style better so I switched.